server/sonar-server/src/main/java/org/sonar/server/es/textsearch/ComponentTextSearchFeature.java

```diff
@@ -17,127 +17,26 @@
 
-import static org.elasticsearch.index.query.QueryBuilders.boolQuery;
-import static org.elasticsearch.index.query.QueryBuilders.matchQuery;
-import static org.elasticsearch.index.query.QueryBuilders.termsQuery;
-import static org.sonar.server.es.DefaultIndexSettingsElement.SEARCH_GRAMS_ANALYZER;
-import static org.sonar.server.es.DefaultIndexSettingsElement.SEARCH_PREFIX_ANALYZER;
-import static org.sonar.server.es.DefaultIndexSettingsElement.SEARCH_PREFIX_CASE_INSENSITIVE_ANALYZER;
-import static org.sonar.server.es.DefaultIndexSettingsElement.SORTABLE_ANALYZER;
+public interface ComponentTextSearchFeature {
 
-public enum ComponentTextSearchFeature {
+  enum UseCase {
+    GENERATE_RESULTS, CHANGE_ORDER_OF_RESULTS;
+  }
 
-  EXACT_IGNORE_CASE {
-    @Override
-    public QueryBuilder getQuery(ComponentTextSearchQuery query) {
-      return matchQuery(SORTABLE_ANALYZER.subField(query.getFieldName()), query.getQueryText())
-        .boost(2.5f);
-    }
-  },
-  PREFIX {
-    @Override
-    public QueryBuilder getQuery(ComponentTextSearchQuery query) {
-      return prefixAndPartialQuery(query.getQueryText(), query.getFieldName(), SEARCH_PREFIX_ANALYZER)
-        .boost(3f);
-    }
-  },
-  PREFIX_IGNORE_CASE {
-    @Override
-    public QueryBuilder getQuery(ComponentTextSearchQuery query) {
-      String lowerCaseQueryText = query.getQueryText().toLowerCase(Locale.getDefault());
-      return prefixAndPartialQuery(lowerCaseQueryText, query.getFieldName(), SEARCH_PREFIX_CASE_INSENSITIVE_ANALYZER)
-        .boost(2f);
-    }
-  },
-  PARTIAL {
-    @Override
-    public QueryBuilder getQuery(ComponentTextSearchQuery query) {
-      BoolQueryBuilder queryBuilder = boolQuery();
-      split(query.getQueryText())
-        .map(text -> tokenQuery(text, query.getFieldName(), SEARCH_GRAMS_ANALYZER))
-        .forEach(queryBuilder::must);
-      return queryBuilder
-        .boost(0.5f);
-    }
-  },
-  KEY {
-    @Override
-    public QueryBuilder getQuery(ComponentTextSearchQuery query) {
-      return matchQuery(SORTABLE_ANALYZER.subField(query.getFieldKey()), query.getQueryText())
-        .boost(50f);
-    }
-  },
-  RECENTLY_BROWSED {
-    @Override
-    public Stream<QueryBuilder> getQueries(ComponentTextSearchQuery query) {
-      Set<String> recentlyBrowsedKeys = query.getRecentlyBrowsedKeys();
-      if (recentlyBrowsedKeys.isEmpty()) {
-        return Stream.empty();
-      }
-      return Stream.of(termsQuery(query.getFieldKey(), recentlyBrowsedKeys).boost(100f));
-    }
-  },
-  FAVORITE {
-    @Override
-    public Stream<QueryBuilder> getQueries(ComponentTextSearchQuery query) {
-      Set<String> favoriteKeys = query.getFavoriteKeys();
-      if (favoriteKeys.isEmpty()) {
-        return Stream.empty();
-      }
-      return Stream.of(termsQuery(query.getFieldKey(), favoriteKeys).boost(1000f));
-    }
+  default UseCase getUseCase() {
+    return UseCase.GENERATE_RESULTS;
   };
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this empty statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)